### PR TITLE
adds social share functions

### PR DESCRIPTION
**Pull request by akash1810**
reads social config from sheet. I've created a `tempSocialConfig` as I don't have the right janus access or google sheet access. I'm not set up to run this project locally, so these changes may not work (though they look like they would...).

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -4,19 +4,31 @@ import share from './lib/share';
...
     sheetToDOM(config.sheetId, config.sheetName, builder, function callback(resp){
-        [].slice.apply(builder.querySelectorAll('.interactive-share')).forEach(shareEl => {
-            var network = shareEl.getAttribute('data-network');
-            shareEl.addEventListener('click', () => shareFn(network));
-        });
         const youTubeId = resp.sheets[config.sheetName][0].youTubeId;
         const youTubeTrailerId = resp.sheets[config.sheetName][0].youTubeTrailerId;
+
+        const socialConfig = resp.sheets[config.socialSheetName];
+        addShareHandler(builder.querySelectorAll('.interactive-share'), socialConfig);
```
Comment (akash1810) [inline src/js/main.js]: `socialConfig` is an `Object`, pass it straight through as `addShareHandler` uses [object destructuring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring).

File: src/js/main.js [hunk 2]
```diff
@@ -4,19 +4,31 @@ import share from './lib/share';
 import sheetToDOM from './lib/sheettodom';
 import emailsignupURL from './lib/emailsignupURL';
 
-var shareFn = share('Interactive title', 'http://gu.com/p/URL', '#Interactive');
+// TODO remove once information is in sheet
+const tempSocialConfig = {
+    title: 'Interactive title',
+    url: 'http://gu.com/p/URL',
+    hashtag: '#Interactive',
+    facebookImage: 'https://media.guim.co.uk/4bf1338c1496f82606c7c0bc41274e22cfe96425/358_294_3280_1968/1000.jpg',
```
Comment (gidsg) [inline src/js/main.js]: I would be tempted to refactor the `share.js` lib, to make the image parameters optional so it just falls back to the `og:image` provided by Composer/Frontend. I think all this is really geared to where they want a custom share CTA which isn't needed for that project.
However, if that proves tricky, I don't mind the current approach.

File: src/js/main.js [hunk 3]
```diff
@@ -4,19 +4,31 @@ import share from './lib/share';
 import sheetToDOM from './lib/sheettodom';
 import emailsignupURL from './lib/emailsignupURL';
 
-var shareFn = share('Interactive title', 'http://gu.com/p/URL', '#Interactive');
+// TODO remove once information is in sheet
+const tempSocialConfig = {
+    title: 'Interactive title',
+    url: 'http://gu.com/p/URL',
```
Comment (gidsg) [inline src/js/main.js]: just use `window.location`?

Comment (akash1810) [inline src/js/main.js]: short urls are needed for twitter's 140 char limit though

File: src/js/main.js (hunk 2 above)
Comment (akash1810) [inline src/js/main.js]: sounds sensible, will take a look at this a bit later

File: src/js/main.js (hunk 3 above)
Comment (gidsg) [inline src/js/main.js]: I believe not anymore, as twitter auto shortens them using t.co, Frontend no longer uses the `shortUrl` see https://github.com/guardian/frontend/pull/13568

Comment (akash1810) [inline src/js/main.js]: ❤️ awesome!

Comment (akash1810) [inline src/js/main.js]: updated now

**Comment by gidsg** [comment]
closing in favour of #50

**Pull request closed without merging**